prompt.sh

```
@@ -1,32 +1,328 @@
+  'rm' -f "$gs_done_file"
+
+  # If the MONITOR option is set, we need to unset it, to stop zsh from spamming four job info messages!
+  # We do this in a subshell so we won't affect the setting in the outer (user's) shell.
+  # CONSIDER: Instead of subshell, we could check the value of MONITOR before, and restore it afterwards.
+  (
+    [[ -n "$ZSH_NAME" ]] && unsetopt MONITOR
+    # Start running the git status process in the background
+    # -uall lists files below un-added folders; without it only the parent folder is listed
+    ( git status --porcelain -uall 2> /dev/null > "$gs_porc_file" ; touch "$gs_done_file" ) &
+  )
+  local gs_shell_pid="$!"
```

Shellcheck (0.11.0)

[warning] gs_shell_pid appears unused. Verify use (or export if used externally).

(SC2034)